File: android/entities/src/main/java/com/tiagohs/entities/contents/ContentLinkScreen.kt
package com.tiagohs.entities.contents

import com.google.gson.annotations.SerializedName
import com.tiagohs.entities.click.Click
import com.tiagohs.entities.image.Image
import java.io.Serializable

class ContentLinkScreen(
    @SerializedName("title")
    val title: String? = null,

    @SerializedName("subtitle")
    val subtitle: String? = null,

    @SerializedName("description")
    val description: String,

    @SerializedName("image")
    val image: Image? = null,

    @SerializedName("click")
    val click: Click? = null
): Content(), Serializable

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/Contents/ContentLinkScreen.swift
//
//  ContentLinkScreen.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 21/02/21.
//

import Foundation
import ObjectMapper

class ContentLinkScreen: Content {
    var title: String?
    var subtitle: String?
    var description: String!
    var image: LocalImage?
    var click: Click?
    
    required init?(map: Map) {
        super.init(map: map)
    }

    override func mapping(map: Map) {
        title                                           <- map["title"]
        subtitle                                        <- map["subtitle"]
        description                                     <- map["description"]
        image                                           <- map["image"]
        click                                           <- map["click"]
    }
}
